### PR TITLE
Add @typing_extensions.deprecated

### PR DESCRIPTION
**Pull request by JelleZijlstra**
See PEP 702, python/peps#2942. This should not be merged until the PEP
is merged into the repo.

### CODE REVIEW

File: src/typing_extensions.py [hunk 1]
```diff
@@ -2129,6 +2130,47 @@ def method(self) -> None:
...
+            @deprecated("Use B instead")
+            class A:
+                pass
+
+            @deprecated("Use g instead")
+            def f():
+                pass
+
+            @deprecated("int support is deprecated")
+            @overload
```
Comment (JelleZijlstra) [inline src/typing_extensions.py:2152]:
```suggestion
            @overload
            @deprecated("int support is deprecated")
```

File: src/typing_extensions.py [hunk 2]
```diff
@@ -2129,6 +2130,47 @@ def method(self) -> None:
...
+            def g(x: int) -> int: ...
+            @overload
+            def g(x: str) -> int: ...
+
+        When this decorator is applied to an object, the type checker
+        will generate a diagnostic on usage of the deprecated object.
+
+        No runtime warning is issued. The decorator sets the ``__deprecated__``
+        attribute on the decorated object to the deprecation message
+        passed to the decorator.
```
Comment (hauntsaninja) [inline src/typing_extensions.py:2162]: Should this more clearly call out the behaviour of `__deprecated__` with overloads?

**Review by hauntsaninja — COMMENTED** [review]
Looks reasonable, just had one question

File: src/typing_extensions.py (hunk 2 above)
Comment (JelleZijlstra) [reply src/typing_extensions.py]: How is it now?

**Review by hauntsaninja — APPROVED** [review]
Thanks! The new words looks good to me.

**Pull request merged**